private-tls.yml

```diff
@@ -0,0 +1,282 @@
+---
+# ------------------------------------------------
+# Create Private Self-Signed Certificate Infrastructure
+# for intra-cluster TLS traffic on Root Host
+# ------------------------------------------------
+- name: Private TLS
+  hosts:
+    - db
+
+  pre_tasks:
+    - name: Define CA variables
+      set_fact:
+        openssl_ca_common_name: miarec-CA
+        openssl_ca_key: "{{ openssl_ca_common_name }}.key"
+
+        openssl_base_dir: /etc/openssl
+        openssl_ca_dir: "{{ openssl_base_dir }}/ca"
+        openssl_servers_dir: "{{ openssl_base_dir }}/servers"
```

Comment:
@timothylytle Can we use the following flow:

1. With Terraform, we generate CA certificate (private key and certificate). The will be stored in terraform state
2. Terraform will export the CA certificate as local files on `terraform apply`
3. Ansible playbook will push the local CA certificates to each host to establish trust
4. Ansible playbook will generate individual SSL certificate for each server/client application and sign it with CA certificate.

Terraform there is optional. User can generate CA certificate locally with openssl command if necessary.

When we store CA private key on the DB server, we make it a good target for attack. If the DB server is compromised, then the atacker can sign own TLS certificates with CA key and connect to other protected servers easily.

I found the following helful Terraform module: https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/self_signed_cert

Another improvement:

- Imagine a customer wants to use own CA infrastructure. They don't want to share CA private key with anyone. But they can generate a bunch of private/public keys for each application and sign with their CA.
- In this case, Ansible playbook will simply copy the existing private/public keys to each server rather than generating/signing them.

Reply:
@genben  This makes sense, this flow should be possible, let me do some testing with that terraform module to confirm. if so I will update.

For the improvement, this is possible now, if the key/cert already existing on the db host, we can supply ansible the path,  by supplying the below variables and skipping `private-tls` playbook. 
 - `redis_tls_cert` 
 - `redis_tls_key`
 - `postgresql_ssl_cert_file`
 - `postgresql_ssl_key_file`
 - `pgbouncer_client_tls_cert_file`
 - `pgbouncer_client_tls_key_file`

If it does not exist on the db host, the step that moves the files would still need to be executed, I will look into this also